shared/actions/favorite.js

```diff
@@ -164,52 +164,40 @@ function _getFavoritesRPCToFolders(
   return folders
 }
 
-function* _addSaga(action: FavoriteGen.FavoriteAddPayload): Saga.SagaGenerator<any, any> {
+function* _addOrIgnoreSaga(
```

Comment:
Why can't this be a pure fn? You can either use your error handler or CallAndWrap